python-threatexchange/threatexchange/fetcher/fetch_state_base.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python3
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+"""
+Base classes for passing data between SignalExchangeAPIs and other interfaces.
+
+Many implementations will choose to extend these to add additional metadata
+needed to power API features.
+"""
+
+
+from dataclasses import dataclass
+from enum import Enum
+import typing as t
+
+from threatexchange.signal_type.signal_base import SignalType
+
+
+TFetchStateCheckpoint = t.TypeVar("TFetchStateCheckpoint")  # TODO for now
+
+
+class SignalOpinionCategory(Enum):
+    """
+    What the opinion on a signal is.
+
+    Some APIs may not support all of these, but each of these should influence
+    what action you might take as a result of matching, otherwise it might
+    make more sense as a tag
+    """
+
+    FALSE_POSITIVE = 0  # Signal generates false positives
+    WORTH_INVESTIGATING = 1  # Indirect indicator
+    TRUE_POSITIVE = 2  # Confirmed meets category
+
+
+@dataclass
+class SignalOpinion:
+    """
+    The metadata of a single signal upload.
+
+    Certain APIs won't have any concept of owner, category, or tags,
+    in which case owner=0, category=TRUE_POSITIVE, tags=[] is reasonable
+    default.
+    """
+
+    owner: int
```

Comment:
nit/ignore: int vs string?

Reply:
Had str, changed to int because I thought it might work , might have to change it back to str depending on implementation.